Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM fedora:latest
+
+# install dependencies
+WORKDIR /opt
+RUN dnf update -y && \
+    dnf install -y cmake gcc-c++ make openssl-devel zlib-devel libuv-devel git && \
+    git clone https://github.com/hoytech/uWebSockets.git && \
+    cd uWebSockets && \
+    make && \
+    make install && \
+    cd .. && \
+    dnf clean all
+# setup repo
+WORKDIR /app
+COPY . /app
+RUN git submodule update --init --recursive && \
+    rm -rf build
+# build the project
+RUN mkdir build && \
+    cd build && \
+    cmake .. -DBUILD_EXAMPLES=On && \
+    make
+# run the example
+ENTRYPOINT ["sh", "-c", "./build/example$EXAMPLE_NUMBER"]
```

Comment:
Maybe we can pin it to a certain example instead of leaving it as `EXAMPLE_NUMBER`. Otherwise I think a simple docker run might not work?

Reply:
Or maybe we can tweak the README to include the ENV that is required here

Reply:
Have added to README already.